braze/client.py

```diff
@@ -106,6 +108,60 @@ def _post_request(self, body, endpoint):
 
             raise BrazeClientError from exc
 
+    def _get_request(self, params, endpoint):
+        """
+        Send a GET request with the specified parameters and endpoint.
+
+        Arguments:
+            params (dict): The request parameters.
+            endpoint (str): The endpoint for the API e.g. /messages/send.
+        Returns:
+            resp (json): The HTTP response in JSON format.
+        Raises:
+            BrazeClientError: If a failure message is returned.
+            BrazeBadRequestError: If a 400 status code is returned.
+            BrazeUnauthorizedError: If a 401 status code is returned.
+            BrazeForbiddenError: If a 403 status code is returned.
+            BrazeNotFoundError: If a 404 status code is returned.
+            BrazeRateLimitError: If a 429 status code is returned.
+            BrazeInternalServerError: If a 5XX status code is returned.
+        """
+        self.session.headers.update(
+            {"Authorization": f"Bearer {self.api_key}", "Content-Type": "application/json"}
+        )
+
+        resp = self.session.get(urljoin(self.api_url, endpoint), params=params, timeout=2)
+
+        try:
```

Comment:
Everything from here down is the same as the `_post_request` Could we dry this up a bit please?

Reply:
Sure

Reply:
I am not currently implementing retries in my command, but retries may be necessary in cases where the API does not return the expected result due to certain exceptions, while the data may still be present.

Reply:
I am considering combining the code into a single method based on a specific condition to avoid duplication.